crates/ironrdp-cliprdr-format/src/bitmap.rs

```diff
@@ -709,35 +703,43 @@ fn decode_png(mut input: &[u8]) -> Result<(png::OutputInfo, Vec<u8>), BitmapErro
     Ok((info, buffer))
 }
 
-/// Convert PNG to `CF_DIB` format.
+/// Converts PNG to `CF_DIB` format.
 pub fn png_to_cf_dib(input: &[u8]) -> Result<Vec<u8>, BitmapError> {
-    let (info, input_buffer) = decode_png(input)?;
-    let (header, output_buffer) = transform_png(info, input_buffer)?;
+    // FIXME(perf): it’s possible to allocate a single array and to directly write both the header and the actual bitmap inside.
+    // Currently, the code is performing three allocations: one inside `decode_png`, one inside `top_down_rgba_to_bottom_up_bgra`
+    // and one in the body of this function.
```

Comment:
FYI @pacmancoder, we could remove one "big" allocation.
Performance seems good enough so far so I didn’t went ahead with that